src/lib/optitype.ml

```diff
@@ -0,0 +1,31 @@
+
+open Common
+open Run_environment
+
+let hla_type ~work_dir ~run_with ~r1 ~r2 ~run_name nt =
+  let tool = Machine.get_tool run_with (`Biopamed "optitype") in
+  let r1pt = Filename.quote r1#product#path in
+  let r2pt = Filename.quote r2#product#path in
+  let name = sprintf "optitype-%s" run_name in
+  let make =
+    Machine.run_program run_with ~name
+      KEDSL.Program.(Tool.init tool
+        && exec ["mkdir"; "-p"; work_dir]
+        && exec ["cd"; work_dir]
+        && sh "cp -r $(opam config var lib)/optitype/data ."                         (* HLA reference data *)
+        && sh "cp -r $(opam config var lib)/optitype/config.ini.example config.ini"  (* config example *)
```

Comment:
I think this should be part of the `Tool.init` (or at least the `Tool.init` should give something like `export OPTITYPE_DATA=$(opam config var lib)/optitype/` than can be used in the body of the `~make`) because the user may not be using `opam` to provide the tool when they build their `Machine.t`.